python/pyspark/sql/column.py

```diff
@@ -151,6 +162,8 @@ def __init__(self, jc):
     __rdiv__ = _reverse_op("divide")
     __rtruediv__ = _reverse_op("divide")
     __rmod__ = _reverse_op("mod")
+    __pow__ = _bin_func_op("pow")
```

Comment:
That make sense, but _bin_func_op expect that `other` should be Column or float, it's not in a shape that we could easily reused for others.